vsintegration/src/FSharp.Editor/Classification/ClassificationDefinitions.fs

```diff
@@ -0,0 +1,96 @@
+
+namespace Microsoft.VisualStudio.FSharp.Editor
+
+open System
+open System.ComponentModel.Composition
+open System.Windows.Media
+
+open Microsoft.VisualStudio.Language.StandardClassification
+open Microsoft.VisualStudio.Text.Classification
+open Microsoft.VisualStudio.Utilities
+open Microsoft.CodeAnalysis.Classification
+
+open Microsoft.FSharp.Compiler.SourceCodeServices
+
+[<RequireQualifiedAccess>]
+module internal FSharpClassificationTypes =
+    let [<Literal>] Function = "FSharp.Function"
+    let [<Literal>] MutableVar = "FSharp.MutableVar"
+    let [<Literal>] Printf = "FSharp.Printf"
+    let [<Literal>] ReferenceType = ClassificationTypeNames.ClassName
+    let [<Literal>] Module = ClassificationTypeNames.ModuleName
+    let [<Literal>] ValueType = ClassificationTypeNames.StructName
+    let [<Literal>] Keyword = ClassificationTypeNames.Keyword
+    let [<Literal>] Enum = ClassificationTypeNames.EnumName
+    let [<Literal>] Property = "FSharp.Property"
+
+    let getClassificationTypeName = function
+        | SemanticClassificationType.ReferenceType
+        | SemanticClassificationType.Module -> Module
+        | SemanticClassificationType.ValueType -> ValueType
+        | SemanticClassificationType.Function -> Function
+        | SemanticClassificationType.MutableVar -> MutableVar
+        | SemanticClassificationType.Printf -> Printf
+        | SemanticClassificationType.ComputationExpression
+        | SemanticClassificationType.IntrinsicType -> Keyword
+        | SemanticClassificationType.UnionCase
+        | SemanticClassificationType.Enumeration -> Enum
+        | SemanticClassificationType.Property -> Property
+
+module internal ClassificationDefinitions =
+    [<Export; Name(FSharpClassificationTypes.Function); BaseDefinition("identifier")>]
+    let FSharpFunctionClassificationType : ClassificationTypeDefinition = null
+
+    [<Export; Name(FSharpClassificationTypes.MutableVar); BaseDefinition("identifier")>]
+    let FSharpMutableVarClassificationType : ClassificationTypeDefinition = null
+
+    [<Export; Name(FSharpClassificationTypes.Printf); BaseDefinition("identifier")>]
+    let FSharpPrintfClassificationType : ClassificationTypeDefinition = null
+
+    [<Export; Name(FSharpClassificationTypes.Property); BaseDefinition("identifier")>]
+    let FSharpPropertyClassificationType : ClassificationTypeDefinition = null
+
+    [<Export(typeof<EditorFormatDefinition>)>]
+    [<ClassificationType(ClassificationTypeNames = FSharpClassificationTypes.Function)>]
+    [<Name(FSharpClassificationTypes.Function)>]
+    [<UserVisible(true)>]
+    [<Order(After = PredefinedClassificationTypeNames.Keyword)>]
+    type internal FSharpFunctionTypeFormat() as self =
+        inherit ClassificationFormatDefinition()
+        
+        do self.DisplayName <- SR.FSharpFunctionsOrMethodsClassificationType.Value
+           self.ForegroundColor <- Nullable Colors.Black
```

Comment:
Looks like this causes text to be black in dark theme.
It seems that leaving this unassigned results in automatic color (black for light themes, white for dark) that still can be customized.